jax/_src/numpy/lax_numpy.py

```diff
@@ -1167,10 +1167,8 @@ def broadcast_to(array: ArrayLike, shape: Shape) -> Array:
   return _broadcast_to(array, shape)
 
 
-def _split(op: str, ary: ArrayLike, indices_or_sections: Union[int, ArrayLike],
+def _split(op: str, ary: Array, indices_or_sections: Union[int, ArrayLike],
            axis: int = 0) -> List[Array]:
-  _check_arraylike(op, ary)
-  ary = asarray(ary)
```

Comment:
Since this check must be done by every wrapper that calls this helper, wouldn't it make more sense to continue to do the check just once here?

Reply:
If it's a type checking issue, then you can address it by changing the variable name when you cast to array. e.g.
```python
a = asarray(ary)
```
Then `a` is of type `Array` and you can do things like `a.shape` without mypy flagging it as incorrect.

Reply:
I was kinda scared of that due to creating an extra array, but that does seem cleaner